src/EnergyPlus/Plant/Component.cc

```diff
@@ -68,7 +68,18 @@ namespace DataPlant {
         return false;
     }
 
-    void CompData::simulate(EnergyPlusData &state, bool const FirstHVACIteration, bool &InitLoopEquip, bool const GetCompSizFac)
+    void CompData::initLoopEquip(EnergyPlusData &state, bool const GetCompSizFac)
+    {
+        this->compPtr->onInitLoopEquip(state, this->location);
+        this->compPtr->getDesignCapacities(state, this->location, this->MaxLoad, this->MinLoad, this->OptLoad);
+        this->compPtr->getDesignTemperatures(this->TempDesCondIn, this->TempDesEvapOut);
+
+        if (GetCompSizFac) {
+            this->compPtr->getSizingFactor(this->SizFac);
+        }
+    }
+
```

Comment:
Moved the code from inside the if(InitLoopEquip) to a new function here.

Reply:
Wonderful!